rest/route/admin_routes.go

```diff
@@ -0,0 +1,233 @@
+	ServiceFlags model.APIServiceFlags `json:"service_flags"`
+	model        model.APIAdminSettings
+}
+
+func (h *adminPostHandler) Handler() RequestHandler {
+	return &adminPostHandler{}
+}
+
+func (h *adminPostHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	if err := util.ReadJSONInto(r.Body, h); err != nil {
+		return err
+	}
+	h.model = model.APIAdminSettings{
+		Banner:       h.Banner,
+		ServiceFlags: h.ServiceFlags,
+	}
+	return nil
+}
+
+func (h *adminPostHandler) Execute(ctx context.Context, sc data.Connector) (ResponseData, error) {
+	settings, err := h.model.ToService()
+	if err != nil {
+		if _, ok := err.(*rest.APIError); !ok {
+			err = errors.Wrap(err, "API model error")
+		}
+		return ResponseData{}, err
+	}
+	err = sc.SetAdminSettings(settings.(admin.AdminSettings))
+	if err != nil {
+		if _, ok := err.(*rest.APIError); !ok {
+			err = errors.Wrap(err, "Database error")
+		}
+		return ResponseData{}, err
+	}
+	return ResponseData{
+		Result: []model.Model{&h.model},
+	}, nil
+}
+
+// this manages the /admin/banner route, which allows setting the banner
+func getBannerRouteManager(route string, version int) *RouteManager {
+	bph := &bannerPostHandler{}
+	bannerPost := MethodHandler{
+		PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+		Authenticator:     &SuperUserAuthenticator{},
+		RequestHandler:    bph.Handler(),
+		MethodType:        evergreen.MethodPost,
+	}
+
+	bannerRoute := RouteManager{
+		Route:   route,
+		Methods: []MethodHandler{bannerPost},
+		Version: version,
+	}
+	return &bannerRoute
+}
+
+type bannerPostHandler struct {
+	Banner model.APIString `json:"banner"`
+	model  model.APIBanner
+}
+
+func (h *bannerPostHandler) Handler() RequestHandler {
+	return &bannerPostHandler{}
+}
+
+func (h *bannerPostHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	if err := util.ReadJSONInto(r.Body, h); err != nil {
+		return err
+	}
+	h.model = model.APIBanner{
+		Text: h.Banner,
+	}
+	return nil
+}
+
+func (h *bannerPostHandler) Execute(ctx context.Context, sc data.Connector) (ResponseData, error) {
+	err := sc.SetAdminBanner(string(h.Banner))
+	if err != nil {
+		if _, ok := err.(*rest.APIError); !ok {
+			err = errors.Wrap(err, "Database error")
+		}
+		return ResponseData{}, err
+	}
+	return ResponseData{
+		Result: []model.Model{&h.model},
+	}, nil
+}
+
+// this manages the /admin/service_flags route, which allows setting the service flags
+func getServiceFlagsRouteManager(route string, version int) *RouteManager {
+	fph := &flagsPostHandler{}
+	flagsPost := MethodHandler{
+		PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+		Authenticator:     &SuperUserAuthenticator{},
+		RequestHandler:    fph.Handler(),
+		MethodType:        evergreen.MethodPost,
+	}
+
+	flagsRoute := RouteManager{
+		Route:   route,
+		Methods: []MethodHandler{flagsPost},
+		Version: version,
+	}
+	return &flagsRoute
+}
+
+type flagsPostHandler struct {
```

Comment:
why can't we just read these data into the handler? it feels like if we were going to add a flag, we'd need to add it in 3 places, which seems error prone?

Reply:
I had tried to make the request format cleaner with this, but I think you're right that we need to not have this duplicate struct. It's changed now

Reply:
nice!